### PR TITLE
TRIDENT-3476: add northstar-parse service and refactor common lib

### PR DESCRIPTION
**Pull request by jasklmclane**
sbt "parse/run" works!

### CODE REVIEW

**Comment by derrley** [comment]
This is good progress -- I'll update the README tonight with a picture of the two components and how they work together.

File: parse/src/main/scala/io/jask/svc/northstar/Demux.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+class Demux(consumer: KafkaConsumer[UUID, Array[Byte]], config: Config) {
+  private[this] lazy val log = LoggerFactory.getLogger(classOf[Demux])
+
+  def run(): Unit = {
+    val running = true
+    val topic = config.getString("northstar.consume.topic")
+    val timeout = config.getInt("northstar.consume.timeout")
+    log.info("Worker started on topic: ".concat(topic))
+    consumer.subscribe(Collections.singletonList(topic))
```
Comment (derrley) [inline parse/src/main/scala/io/jask/svc/northstar/Demux.scala]: Prefer all our list manipulation and such to be in scala, and just use .asJava/.asScala to interop. This is a perfect example of scala being more concise: the equivalent of the above^ is `Seq(topic).asJava` -- even with the java conversion it's fewer characters.

File: parse/src/main/scala/io/jask/svc/northstar/Demux.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.slf4j.LoggerFactory
+
+import scala.collection.JavaConverters._
+
+class Demux(consumer: KafkaConsumer[UUID, Array[Byte]], config: Config) {
+  private[this] lazy val log = LoggerFactory.getLogger(classOf[Demux])
+
+  def run(): Unit = {
+    val running = true
```
Comment (derrley) [inline parse/src/main/scala/io/jask/svc/northstar/Demux.scala]: val running means nobody can ever stop it? Might as well just `while(true) {}`

Probably good to make this guy extend runnable so the can be tossed into a separate thread and joined up from the outside.

**Comment by derrley** [comment]
:shipit: after addressing the two above ^ issues.

**Comment by jasklmclane** [comment]
Update PR with requested changes, thanks!

**Pull request merged**